src/main/java/entity/Deck.java

```diff
@@ -70,4 +78,42 @@ public String getName() {
     public String getId() {
         return id;
     }
+
+    public Deck clone() {
+        Deck result = new Deck(name, id, author);
+
+        for (Card card : deck) {
+            result.addCard(card);
+        }
+
+        return result;
+    }
+
+    public Collection<String> getProblems() {
+        ArrayList<String> problems = new ArrayList<>();
+
+        if (deckLength() < 60) {
+            problems.add("Deck is too small: %s/60".formatted(deckLength()));
+        } else if (deckLength() > 60) {
+            problems.add("Deck is too big: %s > 60".formatted(deckLength()));
+        }
+
+        HashMap<Card, Integer> counts = new HashMap<>();
+        for (Card card : deck) {
+            if (!counts.containsKey(card))
+                counts.put(card, 0);
+
+            counts.put(card, counts.get(card) + 1);
+
+            if (counts.get(card) == 5) {
```

Comment:
I think that we should add a case where a person tries to add the same card into their deck